src/DataTrail.tsx

```diff
@@ -232,6 +241,12 @@ export class DataTrail extends SceneObjectBase<DataTrailState> implements SceneO
     }
   }
 
+  private _handlePanelDataRequestEvent(evt: PanelDataRequestEvent) {
+    // Store the panel data and trigger the modal to open
+    this.setState({ panelData: evt.payload });
```

Comment:
```suggestion
    this.setState({ panelData: evt.payload, isAddToDashboardModalOpen: true });
```

The `useEffect` usage below should not be ;)